src/styles/theme.js

```diff
@@ -44,11 +46,43 @@ export const useBackgroundColor = () => {
     DarkTheme.colors.background,
   );
 };
+function checkContrast(c, background, theme) {
+  let color = new Color(c);
+  let contrast = color.contrast(background, 'WCAG21');
+
+  while (contrast < 4.5) {
+    if (theme === 'dark') {
+      color = color.lighten(0.1);
+    } else {
+      color = color.darken(0.1);
+    }
+    contrast = color.contrast(background, 'WCAG21');
+    if (contrast >= 4.5) {
+      break;
+    }
+  }
+
+  return color.toString({ format: 'hex' });
+}
+
+export const usePrimaryColor = () => {
+  const primaryLight = checkContrast(primaryColor, '#FFFFFF', 'light');
+  const primaryDark = checkContrast(primaryColor, '#201E1E', 'dark');
```

Comment:
I think it's better to move these `checkContrast` calls out of the hook, otherwise they will be executed multiple times